### PR TITLE
Feat/zap stage 2 dual farms

### PR DESCRIPTION
**Pull request by ape-fede**
- [] Before opening a pull request, please read ApeSwap's [contributing guidelines](https://github.com/ApeSwapFinance/apeswap-frontend/blob/staging/CONTRIBUTING.md).
- [] Explain what your PR does & please link it to a grant/issue if applicable.
- [] If your PR is for a grant, please provide contact & fund recipient information in the PR.
- [] If your PR is work in progress, open it as a `draft`.

### CODE REVIEW

File: src/components/DualAddLiquidity/components/ZapSwitch.tsx [hunk 1]
```diff
@@ -18,21 +19,29 @@ interface ZapSwitchProps {
...
           {t('ZAP')}
         </Text>
         <Box sx={{ width: '50px' }}>
-          <Switch checked={goZap} onChange={handleZapSwitch} sx={styles.switchStyles} />
+          <Switch
+            checked={goZap}
+            onChange={handleZapSwitch}
+            sx={{
+              ...styles.switchStyles,
+              backgroundColor: isDark ? 'rgba(56, 56, 56, 1)' : 'rgba(241, 234, 218, 1)',
```
Comment (Harambe-Nakamoto) [inline src/components/DualAddLiquidity/components/ZapSwitch.tsx:42]: You dont need to use isDark here you can use 'white3'

File: src/components/DualAddLiquidity/ZapLiquidity.tsx [hunk 2]
```diff
@@ -96,12 +120,34 @@ const ZapLiquidity: React.FC<ZapLiquidityProps> = ({ handleConfirmedTx }) => {
...
+                zap?.pairOut?.pair?.token1?.getSymbol(chainId),
+              )} ${zapIntoProductType === ZapType.ZAP_MINI_APE ? t('Farm') : null}`}
+            </Text>
+            <Box sx={{ width: '50px', marginLeft: '10px' }}>
+              <Switch
+                checked={stakeIntoProduct}
+                onChange={() => handleStakeIntoProduct(!stakeIntoProduct)}
+                sx={{
+                  ...styles.switchStyles,
+                  backgroundColor: isDark ? 'rgba(56, 56, 56, 1)' : 'rgba(241, 234, 218, 1)',
```
Comment (Harambe-Nakamoto) [inline src/components/DualAddLiquidity/ZapLiquidity.tsx:144]: You dont need to use isdark here you can use 'white3'

File: src/components/DualDepositModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,286 @@
+/** @jsxImportSource theme-ui */
```
Comment (Harambe-Nakamoto) [inline src/components/DualDepositModal/index.tsx:1]: This file is pretty large you should try to break it up. You can move the actions into another file and use a hook to set the callbacks

File: src/hooks/useApprove.ts [hunk 4]
```diff
@@ -134,22 +134,17 @@ export const useDualFarmApprove = (lpContract, pid: number) => {
...
-          token: tx.to,
-          cat: 'enable',
-        },
-      })
-      dispatch(updateDualFarmUserAllowances(chainId, pid, account))
-      return tx
-    } catch (e) {
-      console.warn(e)
-      return false
-    }
```
Comment (Harambe-Nakamoto) [inline src/hooks/useApprove.ts:152]: Why remove the try catch?

File: src/views/Dex/Zap/components/ZapLiquidityActions.tsx [hunk 5]
```diff
@@ -67,9 +73,10 @@ const ZapLiquidityActions: React.FC<ZapLiquidityActionsProps> = ({
         <Flex sx={{ width: '100%' }}>
           <>
             <Button
-              onClick={approveCallback}
-              disabled={approval !== ApprovalState.NOT_APPROVED}
```
Comment (Harambe-Nakamoto) [inline src/views/Dex/Zap/components/ZapLiquidityActions.tsx:71]: Whats the reason for not using the approvalCallback?

Comment (ape-fede) [reply src/views/Dex/Zap/components/ZapLiquidityActions.tsx:71]: to set a loading state to the button while the contract is being approved

**Pull request merged**